### PR TITLE
Fix chunk issues

### LINKED ISSUE #609: Chunk loading issues
This is caused by the dimension screen change

https://user-images.githubusercontent.com/13991048/181633706-ec5e3ee6-7598-494c-a1ab-a46ea230fe72.mov

This second video is a world generated on dragonfly
https://streamable.com/8l61ji

### PR DESCRIPTION
**Pull request by Prim69**
Update: this isnt working?

this closes #609

### CODE REVIEW

File: server/session/chunk.go [hunk 1]
```diff
@@ -239,3 +239,21 @@ func (s *Session) trackBlob(hash uint64, blob []byte) bool {
...
... 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0}
```
Comment (JustTalDevelops) [inline server/session/chunk.go:246]: You can generate these on startup using the existing chunk API, for example:
```go
serializedData := chunk.Encode(chunk.New(air, world.Overworld.Range()), chunk.NetworkEncoding)
```

File: server/session/session.go [hunk 2]
```diff
@@ -402,8 +402,22 @@ func (s *Session) handleWorldSwitch(w *world.World) {
...
+	s.writePacket(&packet.ChangeDimension{Dimension: dim, Position: pos})
 	s.writePacket(&packet.StopSound{StopAll: silent})
+
+	chunkX := int32(pos.X()) >> 4
+	chunkZ := int32(pos.Z()) >> 4
+	for x := int32(-1); x <= 1; x++ {
+		for z := int32(-1); z <= 1; z++ {
+			_ = s.conn.WritePacket(&packet.LevelChunk{
+				Position:      protocol.ChunkPos{chunkX + x, chunkZ + z},
+				SubChunkCount: 1,
```
Comment (JustTalDevelops) [inline server/session/session.go:415]: This technically isn't right

File: server/session/chunk.go (hunk 1 above)
Comment (JustTalDevelops) [inline server/session/chunk.go:246]: Would probably name this something like `overworldEmptyChunk`. It's a single chunk anyway

File: server/session/chunk.go [hunk 3]
```diff
@@ -239,3 +239,45 @@ func (s *Session) trackBlob(hash uint64, blob []byte) bool {
...
+type emptyChunkData struct {
+	payload       []byte
+	subChunkCount uint32
+}
+
+// The variables below are automatically generated to save runtime performance when sending empty chunks, and represents
+// the full payload required to send an empty chunk in a specific dimension.
+var overworldEmptyChunk, netherEmptyChunk, endEmptyChunk emptyChunkData
+
+func init() {
```
Comment (JustTalDevelops) [inline server/session/chunk.go:254]: Could maybe space this out a bit.

**Review by JustTalDevelops — APPROVED** [review]
Looks good to me if tested, just one small comment.

**Comment by Sandertv** [comment]
Going to close this in favour of the removal of the dimension change screens. Please open an issue if that still doesn't fix the issue.

**Pull request closed without merging**